content/docs/tech-insights/track-sonarqube/index.md

```diff
@@ -0,0 +1,121 @@
+2. Run the Roadie broker client application to connect Sonarqube with backstage.
+```shell
+docker run \
+  --env SONARQUBE_INSTANCE_URL=https://<sonarqube_address> \
+  --env BROKER_TOKEN=sonarqube \
+  --env BROKER_SERVER_URL=https://<tenant_name>.broker.roadie.so \
+  --env SONARQUBE_TOKEN=<sonarqube_token> \
+  roadiehq/broker:5691884253.43.1-sonarqube
+```
+3. Determine the IP address from where the broker is running and set this IP in the broker allow list.
+To set the IP allow list go to Administration > Settings > Broker enter the IP address and subnet mask using CIDR notation.
+4. Confirm that the broker client by looking at the docker container logs. You should see the following text:
+```text
+successfully established a websocket connection to the broker server
+```
+
+## Create an HTTP Data Source to collect data from Sonarqube
+
+1. Within Roadie, click on Tech Insights -> Data Sources and click "Add Datasource"
+2. Set the name to "Sonarqube", and set the description if you prefer.
+3. Select the HTTP Data Provider, then select the proxy we created earlier.
+4. In the Path Extension enter `sonarqube/api/measures/component?component={{metadata.annotations["sonarqube.org/project-key"]}}&metricKeys=bugs,security_rating,duplicated_lines_density,security_hotspots`
+5. Next select an entity that you know has been configured with the Sonarcloud annotation so that we can test the data extraction in the next step.
+6. Click view. You should see the result of a successful HTTP response from Sonarqube.
+7. Next you can select the fields to extract from the API.
+In the Field extraction section select the "JSON with JSONata syntax" extractor. Then enter "bugs" as the fact name, `$.component.measures[metric = "bugs"].value` as the JSONata query, and select "Integer" type. Add a second fact by clicking "Add Fact", enter "security_hotspots", `$.component.measures[metric = "security_hotspots"].value` and select "Integer" again.
+8. Test the Data Source fact extraction by clicking "Check Facts". You should see something like the following.
+![field-extraction.png](field-extraction.png)
+
+9. In the Applies to selection, set the "Has Annotation" field to "sonarqube.org/project-key"
+10. Click Save.
+11. In the Data Source page, click "Actions" and "Run" to trigger an initial run of the Data Source
+
+## Create checks with the data from the Sonarcloud Data Sources
+
+1. Go to the Checks tab in tech insights and click "Add Check"
+2. Enter the name "No Sonarqube Bugs"
+3. Select the "Sonarqube" Data Source and the "bugs" fact.
+4. Select the "Less than" operator and enter the value "1".
+5. Click "Dry run"
+6. Select the component you configured in the earlier step.
+7. You should see it's check result.
+8. In the Applies to selection, set the "Has Annotation" field to "sonarqube.org/project-key"
+9. Click Save
+10. Create another check for "Security Hotspots"
+
+## Create a Data Source to check the presence of the Entity metadata
```

Comment:
I think you can probably omit all of these extra steps from here down. The goal of this tutorial (as I think of it) is to help the user get tech insights to connect to an on-prem SonarQube instance. They can learn the basics of scorecard creation elsewhere and it keeps this tutorial a bit more focussed.